### PR TITLE
Upgrade poetry version to v.1.1

### LINKED ISSUE #1328: Update to poetry@1.1
We should also change:
```toml
[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

### PR DESCRIPTION
**Pull request by wwarne**
Closes #1328 

Most important changes in 1.1 -  https://python-poetry.org/blog/announcing-poetry-1-1-0.html

### CODE REVIEW

File: {{cookiecutter.project_name}}/pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [build-system]
-requires = ["poetry>=1.0"]
-build-backend = "poetry.masonry.api"
+requires = ["poetry-core>=1.1.0"]
```
Comment (sobolevn) [inline {{cookiecutter.project_name}}/pyproject.toml:2]: We would also need to remove our `importlib-metadata` hacks from packaging files.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot! 

Btw, `1.1.1` is already released. Can you please adjust the version?

File: {{cookiecutter.project_name}}/pyproject.toml (hunk 1 above)
Comment (wwarne) [reply {{cookiecutter.project_name}}/pyproject.toml]: I've changed version and removed `importib-metadata` version pinning.

**Review by sobolevn — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [build-system]
-requires = ["poetry>=1.0.5"]
-build-backend = "poetry.masonry.api"
+requires = ["poetry-core>=1.1.1"]
```
Comment (DmytroLitvinov) [inline pyproject.toml:2]: The latest version of [`poetry-core` ](https://pypi.org/project/poetry-core/#history)is `1.0.0`.
I guess it should be as in [the documentation](https://python-poetry.org/blog/announcing-poetry-1-1-0.html) that you already mentioned in the section **Standalone build backend**.

File: {{cookiecutter.project_name}}/pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [build-system]
-requires = ["poetry>=1.0"]
-build-backend = "poetry.masonry.api"
+requires = ["poetry-core>=1.1.1"]
```
Comment (DmytroLitvinov) [inline {{cookiecutter.project_name}}/pyproject.toml:2]: Same as the previous comment.

**Review by DmytroLitvinov — CHANGES_REQUESTED** [review]

**Comment by DmytroLitvinov** [comment]
@sobolevn, I guess we should regenerate lock file after removing `importlib-metadata` from `pyproject.toml` file?

**Comment by sobolevn** [comment]
@DmytroLitvinov yes! Exactly.

File: pyproject.toml (hunk 2 above)
Comment (wwarne) [reply pyproject.toml]: Yes, you're definitely right. I'm sorry.  

https://github.com/python-poetry/poetry-core 

I'm surprised it worked with 1.1.1. without any warnings or errors.

**Comment by wwarne** [comment]
@DmytroLitvinov @sobolevn I've run `poetry lock` and added updated `poetry.lock` file

**Comment by sobolevn** [comment]
Looks like `$POETRY_CACHE_DIR` has be renamed or removed.

**Comment by wwarne** [comment]
I think it still works because:

https://python-poetry.org/docs/configuration/#using-environment-variables

> The environment variables must be prefixed by POETRY_ and are comprised of the uppercase name of the setting and with dots and dashes replaced by underscore

And below it:
```

**Available settings**
cache-dir: string

The path to the cache directory used by Poetry.
```

I did a little test:

- docker run --rm -it python:3.7.7-slim-buster bash
- installed poetry 1.1.1
- did `export POETRY_CACHE_DIR='/var/cache/pypoetry'` - this directory didn't existed yet

- `poetry init`  and then `poetry add pytest`
after package was installed I checked /var/cache/pypoetry and it was created and contained some .whl files

**Comment by sobolevn** [comment]
@wwarne so, any ideas why do we have a failure here?

**Comment by wwarne** [comment]
> @wwarne so, any ideas why do we have a failure here?

I'm trying to reproduce it on my local machine and I think probably it is related with `poetry.lock` and `importlib-metadata`. 

I have updated poetry.lock file but I did it on template themself and not in `{{cookiecutter.project_name}}`, my bad. 
But even if I update the second .lock file  it still failed to build docker image properly.

Poetry uses `importlib-metadata` but when we are running `poetry install`  it tries to change its version or reinstall it and since we are using `POETRY_VIRTUALENVS_CREATE=false` it is all happens in the same python environment and ends badly.

I'm going to try different things to see what happens.

**Comment by sobolevn** [comment]
Steps:
1. remove these two lines: https://github.com/wemake-services/wemake-django-template/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/pyproject.toml#L11-L12
2. `rm poetry.lock`
3. run `poetry lock`
4. return these two lines back: https://github.com/wemake-services/wemake-django-template/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/pyproject.toml#L11-L12
5. done!

**Comment by wwarne** [comment]
So basically previous build fails were because then you run poetry with `--no-dev` it tries to remove all packages not specified in `dependencies` section of `pyproject.toml`. But it also removes its own requirements and fails to do anything after this.
Discussion about it - https://github.com/python-poetry/poetry/issues/2102
I used suggested workaround with `poetry add --optional poetry` and docker build now finishes without errors.

**Comment by sobolevn** [comment]
But, we don't use `--no-dev` in CI. We use `--build-arg DJANGO_ENV=development`, see `docker-compose.yml`.

**Comment by wwarne** [comment]
I think I mixed-up with different errors.

Before I regenerated `poetry.lock` (and only removed `importlib-metadata` from `pyproject.toml`) CI failed on installing new packages like so - `Unable to find installation candidates for wcwidth (0.2.4)`. For now  I think maybe it was just some problem with `pypy`  because I have tried to run CI on my local machine with the above setup and it passed build stage and run tests, there were no problems with `wcwidth` or any other package. But previously I also tried to build to build production image and failed.
With production image it was easier to understand what were going on so I found about poetry bug with removing its own dependencies.

After that `production build` succeeded and CI build too.  So for now docker can build CI image and production image without errors but CI failed because then I did `poetry lock` it bumped `django-health-check` from `3.12.1` to `3.14.2` and now it is started to check all cache backends and  `axes_cache` is a dummy cache so the checks fails.

What is your ideas about what to do next?

**Comment by sobolevn** [comment]
You can force `poetry` to use `django-health-check = "3.12.1"`

**Comment by sobolevn** [comment]
Let's bump the limit to `750MiB`

**Review by sobolevn — APPROVED** [review]
Thanks a lot! Very helpful! 👍 

And a lot of efforts put into it. Which is highly appreciated!

**Pull request merged**